torch/distributed/_composable/fsdp/_fsdp_collectives.py

```diff
@@ -105,6 +110,129 @@ def foreach_all_gather_copy_out(
         torch._foreach_copy_(out, splits)  # one `copy_` per parameter
 
 
+@torch.no_grad()
+def foreach_reduce_scatter(
+    fsdp_params: List[FSDPParam],
+    unsharded_grads: List[torch.Tensor],
+    group: dist.ProcessGroup,
+    reduce_scatter_stream: torch.cuda.Stream,
+    orig_dtype: torch.dtype,
+    reduce_dtype: Optional[torch.dtype],
+    device: torch.device,
+    predivide_factor: float,
+    postdivide_factor: float,
+) -> torch.cuda.Event:
+    """
+    Args:
+        unsharded_grads (List[torch.Tensor]): This list owns the references to
+            the unsharded gradients computed by autograd, so clearing this list
+            frees the gradients.
+    """
```

Comment:
maybe copy ``Copy-in/Reduce-Scatter/View-Out Example`` to ``foreach_reduce_scatter`` and ``foreach_reduce_scatter_copy_in`` ?